File: 167. Two Sum II - Input array is sorted/solution.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
  let start = 0;
  let end = numbers.length - 1;

  while(end > start) {
    const sum = numbers[start] + numbers[end];
    if (target === sum) break;
    else if (target > sum) start++;
    else end--;
  }
  return [++start, ++end];
};

File: 167. Two Sum II - Input array is sorted/solution.go
func twoSum(numbers []int, target int) []int {
	indexedNumbers := map[int]int{}
	for idx, number := range numbers {
		if value, exists := indexedNumbers[target-number]; exists {
			return []int{value + 1, idx + 1}
		}
		indexedNumbers[number] = idx
	}
	return []int{}
}